python/orca/tutorial/tensorflow/ncf_orca.py

```diff
@@ -0,0 +1,160 @@
+
+    model = keras.Model(inputs=[user, item], outputs=outputs)
+    model.compile(optimizer= "adam",
+                  loss= "binary_crossentropy",
+                  metrics=['accuracy'])
+    return model
+
+cluster_mode = str(sys.argv[1])
+# cluster_mode = "k8s"
+if cluster_mode == "local":
+    sc = init_orca_context(memory=str(sys.argv[2]))
+elif cluster_mode == "standalone":
+    sc = init_orca_context("standalone", cores=int(sys.argv[3]), num_nodes=int(sys.argv[4]), memory=str(sys.argv[2]))
+elif cluster_mode == "yarn":
+    sc = init_orca_context("yarn", cores=int(sys.argv[3]), num_nodes=int(sys.argv[4]), memory=str(sys.argv[2]))
+elif cluster_mode == "k8s":
+    sc = init_orca_context(cluster_mode="k8s", cores=int(sys.argv[3]), num_nodes=int(sys.argv[4]), memory=str(sys.argv[2]),
+                           master="k8s://https://172.16.0.200:6443",
+                           container_image="10.239.45.10/arda/intelanalytics/bigdl-k8s-spark-3.1.2:0.14.0-SNAPSHOT",
+                           conf={"spark.kubernetes.driver.volumes.persistentVolumeClaim.nfsvolumeclaim.options.claimName": "nfsvolumeclaim",
+                                 "spark.kubernetes.driver.volumes.persistentVolumeClaim.nfsvolumeclaim.mount.path": "/bigdl2.0/data",
+                                 "spark.kubernetes.executor.volumes.persistentVolumeClaim.nfsvolumeclaim.options.claimName": "nfsvolumeclaim",
+                                 "spark.kubernetes.executor.volumes.persistentVolumeClaim.nfsvolumeclaim.mount.path": "/bigdl2.0/data"})
+elif cluster_mode == "spark-submit":  # To test k8s using spark-submit
+    sc = init_orca_context(cluster_mode="spark-submit")
+
+
+data_path = "./bigdl2.0/data"
+data_type = "ml-1m"
+# Need spark3 to support delimiter with more than one character.
+start = time.time()
+spark= OrcaContext.get_spark_session()
+from pyspark.sql.types import StructField, StructType, IntegerType
+
+schema = StructType(
+    [
+        StructField('_c0', IntegerType(), True),
+        StructField('_c1', IntegerType(), True)
+    ]
+)
+df0 = spark.read.csv("{}/{}/ratings.csv".format(data_path, data_type), sep="::",schema=schema,
+                     header=False)
+df0 = df0.withColumnRenamed('_c0','user').withColumnRenamed('_c1','item')
+#print(int(df.describe("item").filter("summary = 'max'").select("item").first().asDict()['item']))
+min_user_id = df0.agg({"user": "min"}).collect()[0]["min(user)"]
+max_user_id = df0.agg({"user": "max"}).collect()[0]["max(user)"]
+min_item_id = df0.agg({"item": "min"}).collect()[0]["min(item)"]
+max_item_id = df0.agg({"item": "max"}).collect()[0]["max(item)"]
+print(min_user_id, max_user_id, min_item_id, max_item_id)
+from pyspark.sql import functions
+df = df0.withColumn('label', functions.lit(1))
+
+from pyspark.sql import functions as F
+from pyspark.sql import types as T
+import random
+
+def neg_sample(x):
+    item_count = len(x) * 4
```

Comment:
The number of negative samples for each (user, item) pair should be modifiable, and you can specify it in arguments as well.